storages/backends/gcloud.py

```diff
@@ -142,10 +144,18 @@ def get_default_settings(self):
             # roll over.
             "max_memory_size": setting("GS_MAX_MEMORY_SIZE", 0),
             "blob_chunk_size": setting("GS_BLOB_CHUNK_SIZE"),
+            "sa_email": setting("GS_SA_SIGNING_EMAIL")
         }
 
     @property
     def client(self):
+        credentials, project_id = auth.default(scopes=['https://www.googleapis.com/auth/cloud-platform'])
+        if self.project_id is None:
+            self.project_id = project_id
+        credentials.refresh(requests.Request())
```

Comment:
Do we need to refresh the credentials every time we access `client`? Basically I'm wondering if we can follow the example already there where we only create the client once. The only reason not to guard against this in what you've got is the refresh. Without it we end up with something such as:

```python
if self._client is None:
    project_id, credentials = self.project_id, self.credentials
    if project_id is None and credentials is None:
       credentials, project_id = auth.default(scopes=['https://www.googleapis.com/auth/cloud-platform'])
       if not hasattr(credentials, "service_account_email"):
           credentials.service_account_email = self.sa_email
      self._client = Client(project=project, credentials=credentials)
```

Reply:
Happy to add this. I just wanted to refresh the client if it has been a long time since the user called it.